### DISCUSSION TITLE
Is there any way to get to know from code whether we run as a worker launched with `manage.py qcluster`?

### DISCUSSION THREAD

**Discussion by pktiuk** [django-q2/django-q2#248] [Q&A] [answered]
I want to run my djqngo-q2 cluster on a separate machine with different hardware, set of libraries and tools, so I would like to limit importing some libs to qcluster workers.        

Is there any simple way of getting to know whether code is running inside of cluster (instead of regular django web server)?

this is example of usage

```py

if is_running_inside_qcluster():
    import some_lib
    import antoher_lib_not_available_on_other_machines
    import cuda_processing


def processing():
    if is_running_inside_qcluster():
        cuda_processing()
    else:
        regular_processing()
```
 (I would be also interested in writing some system checks only for clusters)

**Comment by GDay** [comment]
I don't think there is, but maybe you could set an environment variable on that second machine and just check if that's there?

**Reply by pktiuk** [reply]
Yes, I can and it seems, that it will end up doing this, but I hoped to find something more reliable and elegant.

**Comment by pktiuk** [comment, marked as answer]
I decided to deal with it by checking arguments

```
import sys

def is_in_qcluster() -> bool:
    return "qcluster" in sys.argv
```
